### PR TITLE
chore: Add more tests and documentation

### PR DESCRIPTION
**Pull request by ebebbington**
## Summary

Adds full tests and documentation

* Adds all unit tests for source files
* Adds tests for the advanced tutorial in the docs
* Adds integration tests
* Adds more documentation
* Adds workflows

### CODE REVIEW

**Comment by ebebbington** [comment]
Issue for adding docs: https://github.com/drashland/website/pull/375

**Comment by ebebbington** [comment]
actually this is ready for review

File: README.md [hunk 1]
```diff
@@ -3,99 +3,45 @@
...
+  <a href="https://github.com/drashland/line/actions">
+    <img src="https://img.shields.io/github/workflow/status/drashland/line/master?label=ci">
+  </a>
+  <a href="https://discord.gg/SgejNXq">
+    <img src="https://img.shields.io/badge/chat-on%20discord-blue">
+  </a>
+  <a href="https://twitter.com/drash_land">
+    <img src="https://img.shields.io/twitter/url?label=%40drash_land&style=social&url=https%3A%2F%2Ftwitter.com%2Fdrash_land">
+  </a>
+  <!-- <a href="https://rb.gy/vxmeed">
```
Comment (saragee3) [inline README.md:19]: remove?

File: src/subcommand.ts [hunk 2]
```diff
@@ -12,12 +12,12 @@ export class Subcommand {
   /**
    * This subcommand's description.
    */
-  public description: string = "";
+  public description = "";
```
Comment (saragee3) [inline src/subcommand.ts:15]: why are the types removed?

File: tests/unit/line_test.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    try {
+      Deno.writeFileSync(file, encoder.encode(contents));
+      console.log("Successfully wrote file.");
+    } catch (error) {
+      console.log(error);
+    }
+  }
+}
+
+// const service = new Line({
```
Comment (saragee3) [inline tests/unit/line_test.ts:43]: 👀

File: README.md (hunk 1 above)
Comment (ebebbington) [reply README.md:19]: Sure :)

File: src/subcommand.ts (hunk 2 above)
Comment (ebebbington) [reply src/subcommand.ts:15]: Linter pulled this up, throws errors for properties that explicitly have the type even when it’s inferred

File: tests/unit/line_test.ts (hunk 3 above)
Comment (ebebbington) [reply tests/unit/line_test.ts:43]: Want me to remove it?:p tbh it’s dead code anyway

Comment (ebebbington) [reply tests/unit/line_test.ts]: https://github.com/drashland/line/pull/1/commits/19e714e5f6ff45ed6a10dffd64737c64a868e8a6

File: src/subcommand.ts (hunk 2 above)
Comment (crookse) [reply src/subcommand.ts:15]: yea the linter disagrees with strongly typed, inferred variables like this

**Review by crookse — APPROVED** [review]
LGTM, but let's wait for @saragee3

**Review by saragee3 — APPROVED** [review]
👍

**Pull request merged**